ember-resources/src/core/class-based/resource.ts

```diff
@@ -42,6 +42,8 @@ type ResourceHelperLike<T, R> = InstanceType<
 
 declare const __ResourceArgs__: unique symbol;
 
+type Context = object;
```

Comment:
As an attempt to work with more specific types, we (Jan and me) tried to give the `Context` its own type. That is, we started with `unknown` but stopped with `object`, the reason is, that `destroyable` accepts an `object` at https://github.com/NullVoxPopuli/ember-resources/blob/6fc67cb91466ffd1bfcaa2fdee594fdaf7c9dcfc/ember-resources/src/util/map.ts#L170

Maybe `Context` can be such a type:

```ts
type Context = Exclude<object, Thunk>;
```